File: PRACTICA3/Pila.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        arreglo = new long[n]; 
        top = -1;              
    }

    public void push(long e) throws Exception {
        if (!isFull()) {
            top++;
            arreglo[top] = e;
        } else {
            throw new Exception("Pila llena");
        }
    }

    public long pop() throws Exception {
        if (!isEmpty()) {
            long elemento = arreglo[top];
            top--;
            return elemento;
        } else {
            throw new Exception("Pila vacía");
        }
    }

    public long peek() throws Exception {
        if (!isEmpty()) {
            return arreglo[top];
        } else {
            throw new Exception("Pila vacía");
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        try {
            Pila pila = new Pila(5);
            pila.push(10);
            pila.push(20);
            System.out.println("-----------------");
            System.out.println(pila.peek()); 
            System.out.println(pila.pop());   
            System.out.println(pila.isEmpty()); 
        } catch (Exception e) {
            System.out.println("-----------------");
            System.out.println(e.getMessage());
        }
    }
}

File: PRACTICA3/Pila.py
class Pila:
    def _init_(self, n):
        self.arreglo = [0] * n  
        self.top = -1           
        self.n = n              

    def push(self, e):
        if not self.isFull():
            self.top += 1
            self.arreglo[self.top] = e
            print(f'Elemento {e} añadido a la pila.')  
        else:
            raise Exception("Pila llena")

    def pop(self):
        if not self.isEmpty():
            elemento = self.arreglo[self.top]
            self.top -= 1
            print(f'Elemento {elemento} eliminado de la pila.')  
            return elemento
        else:
            raise Exception("Pila vacía")

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.top]
        else:
            raise Exception("Pila vacía")

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

if __name__ == "_main_":
    print("Ejecutando el programa...")
    pila = Pila(5)  
    pila.push(10)
    pila.push(20)
    print(f'Elemento en la cima: {pila.peek()}')  
    print(f'Elemento retirado: {pila.pop()}')      
    print(f'¿Está la pila vacía? {pila.isEmpty()}') 